docs/calamari/CommunityCollatorsProgram.md

```diff
@@ -0,0 +1,287 @@
+# 🧩  Community Collators Program
+
+## Requirements
+
+1. Hardware Requirement
+
+    a. If running on your own machine, hardware requirement, internet bandwidth (we prefer collators with private node for better decentralization).
+    
+    ```jsx
+    cpu:8 cores
+    memory: 32 GB of ram
+    disk space: 500 GB of disk space
+    bandwidth: 100 Mbps
+    recommended region: EU or US
+    ```
+    
+    b. If running on AWS/Azure/GCP (or other cloud provider), please use an instance type with similar configuration to the following:
+    
+    ```jsx
+    AWS instance: Ubuntu 20.04 (ami-id = ami-04505e74c0741db8d)
+    disk space: r5ad.xlarge
+    recommended region: ca-central, eu-west, eu-central, eu-north, us-east
+    ```
+    
+    NOTE: 
+    
+    - A calamari collator utilizes up to 27gb of ram on a 32gb system.
+    - A calamari collator utilizes up to 25% of cpu on a 24 core system.
+    - A reliable internet connection as well as power supply which should remain uninterrupted to keep local machine **permanently online.**
+
+2. Bonding requirement: `400_000`KMA at least. Make sure your account has more than `400_000`KMA in your free balance.
+
+    You can check candidacy bond on calamari mainnet.
+    ![Candidacy Bond](images/candidacy-bond.png)
+
+## Deployment
+
+### Tips:
+
+1. The node can be deployed by native binary or docker, it depends on your preference.
+    - [Native](#deploy-a-collator-on-ubuntu-using-native-binary)
+    - [Docker](#deploy-a-collator-on-ubuntu-using-docker)
+2. We provide a recent snapshot which can shorten the time on catching up calamari mainnet.
+
+### Deploy a collator on Ubuntu using native binary
+
+We recommend run on Ubuntu **`20.04 LTS`** :
+1. Get latest manta binary from manta repo: [https://github.com/Manta-Network/Manta/releases](https://github.com/Manta-Network/Manta/releases)
+    
+- Get calamari genesis file: [https://github.com/Manta-Network/Manta/blob/manta/genesis/calamari-genesis.json](https://github.com/Manta-Network/Manta/blob/manta/genesis/calamari-genesis.json)
+- Get relaychain(kusama) genesis: [https://github.com/paritytech/polkadot/blob/master/node/service/res/kusama.json](https://github.com/paritytech/polkadot/blob/master/node/service/res/kusama.json)
+
+2. Start a collator by the following command with RPC port enabled. Please specify `your_db_path`.
+
+    > If you want to start collator with snapshot, download snapshot from [here](#download-snapshot),
+    > follow this [section](#native) to deploy node with snapshot.
+
+    ```bash
+    manta --base-path your_db_path \    
+    --name your_collator_name \
+    --chain calamari-genesis.json \
+    --rpc-cors 'all' \
+    --port 30333 \
+    --rpc-port 9933 \
+    --rpc-methods=Unsafe \
+    --force-authoring \
+    --collator \
+    -- \
+    --chain kusama.json
```

Comment:
I still believe the relaychain port should be explicitly specified

I just had a look in the logs and it works without doing so because there is a random-port fallback built in.
Peer discovery between two firewalled nodes on different networks likely won’t work (haven’t tested, but there would have to be some black networking magic (e.g. UPNP)  happening for this to punch through the firewall).
Even if both nodes are in the address book of a non-firewalled bootnode and know where to find each other, their firewalled random port will result in incoming requests being dropped. If both nodes are firewalled, it doesn’t matter which one sends the request, the other one will drop it, never establishing the connection.
Since we’re running the Dotsama relaychain, there’ll be enough non-firewalled nodes to not stall the chain, but it’s a misconfiguration either way. 

```
2022-02-06 23:44:57 [Relaychain] Can't listen on /ip6/::/tcp/30334/ws because: Other(Custom { kind: Other, error: Other(A(A(A(A(Transport(A(Transport(Os { code: 98, kind: AddrInUse, message: "Address already in use" })))))))) })    2022-02-06 23:44:57 [Relaychain] Can't listen on /ip4/0.0.0.0/tcp/30334/ws because: Other(Custom { kind: Other, error: Other(A(A(A(A(Transport(A(Transport(Os { code: 98, kind: AddrInUse, message: "Address already in use" })))))))) })    2022-02-06 23:44:57 [Relaychain] Listening for new connections on 127.0.0.1:39697.
```

Reply:
I tried to start nodes in calamari mainnet. It's fine without opening the port `30334`. This port is open by default.

Reply:
The app _uses_ `30334` by default, but that doesn't mean it's _open_ by default.
If you did this from a standard home network without forwarding `30334` in router I'm sure your node will exhibit the problem i described above that it can peer with nodes it reaches out to, but external nodes can't establish connections with it.
Not sure if there's a way to see outgoing/incoming connections in the node logs, i havent found it.
There's a way to test this by overriding bootnodes with just your new node and trying to spin up a new one from outside your network, this should fail.

I think i'm arguing a technicality here, since functionality-wise as you saw it works without it, but I think our **recommendations** should promote good *network hygiene* and leave it up to the user to ignore it at his own risk.

Reply:
I think this is addressed well enough with the mention that the ports must be *internet accessible* in 
https://github.com/Manta-Network/docs/pull/15/files#diff-7067557063d6480080f02226d66b249be49b52faa348d58f17b06b4deaabccb6R8